avro-builder/src/main/java/com/linkedin/avroutil1/builder/operations/OperationContext.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022 LinkedIn Corp.
+ * Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+
+package com.linkedin.avroutil1.builder.operations;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * common context to all {@link Operation}s run during an execution
+ */
+public class OperationContext {
+
+  private List<Operation> operations = new ArrayList<>(1);
+
+  public void add(Operation op) {
+    if (op == null) {
+      throw new IllegalArgumentException("op required");
+    }
+    operations.add(op);
+  }
+
+  public void run() throws Exception {
+    for (Operation op : operations) {
```

Comment:
This is unsafe -- the caller could turn around and try to call `.add()` on the OperationContext. Similar to the advice in Effective Java (Item 79: Avoid excessive synchronization), better to make a defensive copy of the list here, before calling any plugins.

Reply:
good catch. added a "sealed" flag to separate the lifecycle into a "configure" phase that comes before run()